utils/s3/client.cc

```diff
@@ -1160,42 +1162,63 @@ class client::chunked_download_source final : public seastar::data_source_impl {
                             s3l.warn("Fiber for object '{}' failed: {}. Exiting", _object_name, reply._status);
                             throw httpd::unexpected_status_error(reply._status);
                         }
-                        if (_range == s3::full_range && reply.get_header("Content-Range").empty()) {
+                        if (_range == s3::full_range && !reply.get_header("Content-Range").empty()) {
                             auto content_range_header = parse_content_range(reply.get_header("Content-Range"));
                             _range = range{content_range_header.start, content_range_header.total};
-                            s3l.trace("No range for object '{}' was provided. Setting the range to {} form the Content-Range header",
+                            s3l.trace("No range for object '{}' was provided. Setting the range to {} from the Content-Range header",
                                       _object_name,
                                       _range);
                         }
                         auto in = std::move(in_);
-                        while (_buffers_size < _max_buffers_size && !_is_finished) {
-                            auto start = s3_clock::now();
-                            s3l.trace("Fiber for object '{}' will try to read within range {}", _object_name, _range);
-                            auto buf = co_await in.read();
-                            auto buff_size = buf.size();
-                            gc.read_stats.update(buff_size, s3_clock::now() - start);
-                            _range += buff_size;
-                            _buffers_size += buff_size;
-                            if (buff_size == 0 && _range.length() == 0) {
-                                s3l.trace("Fiber for object '{}' signals EOS", _object_name);
+                        std::exception_ptr ex;
+                        try {
```

Comment:
The try body seem to be just what it was right-shifted? If so, please, split it into two patches, first adds try-catch but doesn't touch the indentation (it becomes broken, but that's fine), second _just_ restores the indentation